sklearn/preprocessing/_encoders.py

```diff
@@ -161,6 +161,19 @@ class OneHotEncoder(_BaseEncoder):
 
         The used categories can be found in the ``categories_`` attribute.
 
+    drop : 'first' or a list/array of shape (n_features,), default=None.
+        Specifies a methodology to use to drop one of the categories per
+        feature. This is useful in situations where perfectly collinear
+        features cause problems, such as when feeding the resulting data
+        into a neural network or an unregularized regression. If only one
+        feature appears in a column, that column will be retained as a
+        column of ones.
+
+        - None : retain all features.
+        - 'first' : drop the first feature in each category.
+        - array : ``drop[i]`` is the value of the feature in ``X[:,i]``
```

Comment:
value of the feature -> category to drop

Reply:
Should it be possible for the user to specify that some features do not drop? Perhaps that's unnecessary complexity, and the current design is good.

Reply:
I added this functionality as well, it was simple to implement.

Reply:
sure--changed to category.